src/Playwright/Transport/Channels/PageChannel.cs

```diff
@@ -117,7 +117,29 @@ internal override void OnMessage(string method, JsonElement? serverParams)
                     FrameDetached?.Invoke(this, serverParams?.GetProperty("frame").ToObject<FrameChannel>(Connection.GetDefaultJsonSerializerOptions()).Object);
                     break;
                 case "dialog":
-                    Dialog?.Invoke(this, serverParams?.GetProperty("dialog").ToObject<DialogChannel>(Connection.GetDefaultJsonSerializerOptions()).Object);
+                    var dialog = serverParams?.GetProperty("dialog").ToObject<DialogChannel>(Connection.GetDefaultJsonSerializerOptions()).Object;
+                    if (Dialog != null)
+                    {
+                        Dialog.Invoke(this, dialog);
+                    }
+                    else
+                    {
+                        if ("beforeunload".Equals(dialog.Type, StringComparison.Ordinal))
+                        {
+                            try
+                            {
+                                await dialog.AcceptAsync(null).ConfigureAwait(false);
+                            }
+                            catch (PlaywrightException)
+                            {
+                                // Noop
```

Comment:
Do we need to log anything. Here if something goes wrong, it will be silent here.